conans/model/conan_file.py

```diff
@@ -108,6 +112,10 @@ def __init__(self, output, runner, settings, conanfile_directory, user=None, cha
         self.env_info = None  # Will be initialized at processing time
         self.deps_env_info = DepsEnvInfo()
 
+        # user declared variables
+        self.user_info = None  # Will be initializad at processing time
+        self.deps_user_info = defaultdict(UserInfo)  # Keys are the package names, and the values a dict with the vars
```

Comment:
No! :) 
It is fine as is. I was wondering, but having a look to ``class DepsEnvInfo(EnvInfo):`` inheriting from EnvInfo, it is not necessary in this case, because all vars are scoped by the package.